### ISSUE TITLE
bug on ida pro custom_script.py

### ISSUE THREAD

**Issue by anaivebird** [qilingframework/qiling#738]
***Describe the bug**
  File "custom_script.py", line 26, in step_hook
    logging.info(f"Executing: {hex(addr)}")
NameError: name 'logging' is not defined
```

from qiling import *

class QILING_IDA():
    def __init__(self):
        pass

    def _show_context(self, ql:Qiling):
        registers = [ k for k in ql.reg.register_mapping.keys() if type(k) is str ]
        for idx in range(0, len(registers), 3):
            regs = registers[idx:idx+3]
            s = "\t".join(map(lambda v: f"{v:4}: {ql.reg.__getattribute__(v):016x}", regs))
            ql.log.info(s)

    def custom_prepare(self, ql:Qiling):
        ql.log.info('Context before starting emulation:')
        self._show_context(ql)

    def custom_continue(self, ql:Qiling):
        ql.log.info('custom_continue hook.')
        self._show_context(ql)
        hook = []
        return hook

    def custom_step(self, ql:Qiling):
        def step_hook(ql, addr, size):
            logging.info(f"Executing: {hex(addr)}")
            self._show_context(ql)

        ql.log.info('custom_step hook')
        hook = []
        hook.append(ql.hook_code(step_hook))
        return hook
    
    def custom_execute_selection(self, ql:Qiling):
        ql.log.info('custom execute selection hook')
        hook = []
        return hook
```

change logging.info(f"Executing: {hex(addr)}") to ql.log.info('Executing: {hex(addr)}') will fix it

**Comment by wtdcode** [comment]
Got it. Will fix ASAP.

________________________________
From: naivebird ***@***.***>
Sent: Monday, March 22, 2021 2:09:06 AM
To: qilingframework/qiling ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [qilingframework/qiling] bug on ida pro custom_script.py (#738)


*Describe the bug
File "custom_script.py", line 26, in step_hook
logging.info(f"Executing: {hex(addr)}")
NameError: name 'logging' is not defined


from qiling import *

class QILING_IDA():
    def __init__(self):
        pass

    def _show_context(self, ql:Qiling):
        registers = [ k for k in ql.reg.register_mapping.keys() if type(k) is str ]
        for idx in range(0, len(registers), 3):
            regs = registers[idx:idx+3]
            s = "\t".join(map(lambda v: f"{v:4}: {ql.reg.__getattribute__(v):016x}", regs))
            ql.log.info(s)

    def custom_prepare(self, ql:Qiling):
        ql.log.info('Context before starting emulation:')
        self._show_context(ql)

    def custom_continue(self, ql:Qiling):
        ql.log.info('custom_continue hook.')
        self._show_context(ql)
        hook = []
        return hook

    def custom_step(self, ql:Qiling):
        def step_hook(ql, addr, size):
            logging.info(f"Executing: {hex(addr)}")
            self._show_context(ql)

        ql.log.info('custom_step hook')
        hook = []
        hook.append(ql.hook_code(step_hook))
        return hook

    def custom_execute_selection(self, ql:Qiling):
        ql.log.info('custom execute selection hook')
        hook = []
        return hook


change logging.info(f"Executing: {hex(addr)}") to ql.log.info('Executing: {hex(addr)}') will fix it

―

**Comment by wtdcode** [comment]
Fixed in b4608d33860193d5831963715304ebd882653e15
